### PR TITLE
Added extra styles for --separate element. Updated padding and margins

### PR DESCRIPTION
**Pull request by Lukewh**
## Done

Reduced aside padding and margins.
Added `.p-toc__item` style and `--separate` modifier (for styling visually separated elements).
Added `--active` modifier.

## QA

1. Pull this branch
2. Run `./run watch`
3. Save https://gist.github.com/Lukewh/dc0f1b9d20a9ef46bb337fac43c03502 to a html file in the parent directory of this repo.
4. Open the saved file and see the toc on the right hand side
5. Scroll up and down and resize the browser

## Details

Fixes: https://github.com/vanilla-framework/vanilla-docs-theme/issues/1.
Needed for new fixed position niceness: https://github.com/ubuntudesign/docs.vanillaframework.io/pull/106

## Screenshots

screen shot 2017-07-18 at 12 08 02
screen shot 2017-07-18 at 12 08 08

### CODE REVIEW

File: scss/_patterns_toc.scss [hunk 1]
```diff
@@ -5,10 +5,28 @@
...
+        margin-bottom: 0;
+      }
+    }
+
     &__link {
       border-bottom: 0;
       color: $color-dark;
-      margin-bottom: .5rem;
+
+      &--active {
```
Comment (anthonydillon) [inline scss/_patterns_toc.scss]: Active is a state. Therefore should be a state class `is-active` https://github.com/canonical-webteam/practices/blob/master/coding/stylesheets.md#state-classes

**Comment by yaili** [comment]
Will there be a demo link?

File: scss/_patterns_toc.scss (hunk 1 above)
Comment (Lukewh) [inline scss/_patterns_toc.scss]: Done

**Comment by Lukewh** [comment]
@yaili there is no demo link, I might take a look and see if I can get it set up.

**Comment by anthonydillon** [comment]
QA issue when you have more than one separated item in the list:
lists - documentation

**Comment by anthonydillon** [comment]
Also, there are some issues running `./run test`.

**Comment by yaili** [comment]
@anthonydillon I don't think you're supposed to have more than the dividers specified in the design. @spencerbygraves ?

**Comment by anthonydillon** [comment]
I think we should. We should be able to split a list more then once and it
to work and look as expected.

**Comment by Lukewh** [comment]
@anthonydillon can you give it another go with the `./run test` please? Also QA again if you get chance.

**Comment by anthonydillon** [comment]
I am getting different spacing between the first split item. Please add `margin-bottom: 0` to all p-toc__item--separate... Also, we should only be using a single pattern class here. Therefore it is either `p-toc__item` or `p-toc__item--separate`

**Pull request merged**